### ISSUE TITLE
Incorrect authentication credentials results in server error

### ISSUE THREAD

**Issue by auahmed** [networknt/light-oauth2#150]
When a user enters the wrong username and password for code authentication, the response should be incorrect credentials rather then server error.

**Comment by stevehu** [comment]
Which authenticator is used? Marketplace?

**Comment by auahmed** [comment]
yes, currently when the user logins with invalid credentials null is returned:
````
logger.error("Failed to authenticate user '" + id + "' with LDAP");
return null;
````

This is also similar to the default authenticator:
````
if(!match) return null;
````

**Comment by stevehu** [comment]
I will take a look this weekend. Thanks for bringing it up.

**Comment by stevehu** [comment]
@auahmed If the credential is not matched, then a null account object is returned and in the basic authentication, the popup window will show up again. This repeats several times depending on the OS configuration and then fail the basic authentication. The Marketplace is using the form and the flow might be different. What is the behavior in your app?

**Comment by auahmed** [comment]
The flow of the app is that the user credentials are passed in the headers to oauth2 as base64-encoded.
As the application is blocking the popup provided by light-oauth2 to use its own form rather, we get an internal server error rather then the popup window to try again. This way it becomes difficult for the user to differentiate if incorrect credentials were provided or any other issue.

**Comment by stevehu** [comment]
Do you have a form to collect the username and password? Is it possible to just customize the marketplace auth? I am working on the light-portal now and will pick up the post authentication soon.

**Comment by auahmed** [comment]
Will check into seeing if its possible to just update the marketplace auth
